collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FieldListUpdateTest.java

```diff
@@ -421,6 +421,22 @@ public void recordingAudio_ShouldChangeRelevanceOfRelatedField() {
                 .assertTextDoesNotExist("Target16");
     }
 
+    @Test
+    public void changeInValueUsedToDetermineIfAQuestionIsRequired_ShouldUpdateTheRelatedRequiredQuestion() {
+        new FormEntryPage("fieldlist-updates")
+                .clickGoToArrow()
+                .clickGoUpIcon()
+                .clickOnGroup("Dynamic required question")
```

Comment:
Could we create a new form for this so we don't need to use the form hierarchy to navigate in the test? I also find the smaller forms make it easier to comprehend what's going on in each test.

Reply:
I was thinking about that but I wanted to have this test in `FieldListUpdateTest` as it's similar to other tests in that class. However `FieldListUpdateTest` is started with that particular form directly so I would either move that test to a separate class or rework `FieldListUpdateTest` to start at the main menu and allow me to choose a form. Do you think it makes sense?

Reply:
Alternatively, we could switch `FieldListUpdateTest` to use `FormEntryActivityTestRule` instead of `BlankFormTestRule`? That would allow each test to set up a different form, but also not require them to go through the full app. For the moment, all the other tests could start with `setUpProjectAndCopyForm` for the current `fieldlist-updates` form and this one test could use a different one.

Reply:
That makes sense but this test is pretty big so let's not block this pr. I've filed a separate issue https://github.com/getodk/collect/issues/6143